src/python/WMCore/MicroService/MSTransferor/MSTransferor.py

```diff
@@ -670,20 +672,22 @@ def makeTransferRucio(self, wflow, dataIn, subLevel, blocks, dataSize, nodes, no
         subLevel = "ALL" if subLevel == "container" else "DATASET"
         dids = blocks if blocks else [dataIn['name']]
 
-        rule = {'copies': 1,
-                'activity': 'Production Input',
-                'lifetime': self.msConfig['rulesLifetime'],
-                'account': self.msConfig['rucioAccount'],
-                'grouping': subLevel,
-                'meta': {'workflow_group': wflow.getWorkflowGroup()},
-                'comment': 'WMCore MSTransferor input data placement'}
+        ruleAttrs = {'copies': 1,
+                     'activity': 'Production Input',
+                     'lifetime': self.msConfig['rulesLifetime'],
+                     'account': self.msConfig['rucioAccount'],
+                     'grouping': subLevel,
+                     'meta': {'workflow_group': wflow.getWorkflowGroup()},
+                     'comment': 'WMCore MSTransferor input data placement'}
+        # add a configurable weight value
+        ruleAttrs["weight"] = self.msConfig['rucioRuleWeight']
```

Comment:
Why not in the `ruleAttrs={}` definition itself?

Reply:
Good question! Let me quickly update it. Thanks